misc/bazel/internal/install.py

```diff
@@ -45,35 +62,59 @@
 
 _WIN_FILE_IN_USE_ERROR_CODE = 32
 
-if destdir.exists() and opts.cleanup:
-    if platform.system() == 'Windows':
+
+def rmdir(dir: pathlib.Path):
+    if platform.system() == "Windows":
         # On Windows we might have virus scanner still looking at the path so
         # attempt removal a couple of times sleeping between each attempt.
         for retry_delay in [1, 2, 2]:
             try:
-                shutil.rmtree(destdir)
+                shutil.rmtree(dir)
                 break
             except OSError as e:
                 if e.winerror == _WIN_FILE_IN_USE_ERROR_CODE:
                     time.sleep(retry_delay)
                 else:
                     raise
         else:
-            shutil.rmtree(destdir)
+            shutil.rmtree(dir)
     else:
-        shutil.rmtree(destdir)
+        shutil.rmtree(dir)
+
+
+if destdir.exists() and opts.cleanup:
+    rmdir(destdir)
+
+
+class RetryException(Exception):
```

Comment:
where is this used? Maybe this should be raised in line #120?

Reply:
It's a leftover, removed.